core/common/src/main/java/alluxio/metrics/EventCounter.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.metrics;
+
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.Level;
+import org.apache.log4j.spi.LoggingEvent;
+
+/**
+ * A log4J Appender that simply counts logging events in three levels:
+ * fatal, error and warn. The class name is used in log4j.properties
+ */
+public class EventCounter extends AppenderSkeleton {
+    private static final int FATAL = 0;
+    private static final int ERROR = 1;
+    private static final int WARN = 2;
+    private static final int INFO = 3;
+    private static EventCounter.EventCounts sCount = new EventCounter.EventCounts();
+
+    /**
+     * the constructor of EventCounter.
+     */
+    public EventCounter() {
+    }
+
+    /**
+     * get the number of fatal log.
+     * @return the number of fatal log
+     */
+    public static long getFatal() {
+        return sCount.get(0);
+    }
+
+    /**
+     * get the number of error log.
+     * @return the number of error log
+     */
+    public static long getError() {
+        return sCount.get(1);
+    }
+
+    /**
+     * get the number of warn log.
+     * @return the number of warn log
+     */
+    public static long getWarn() {
+        return sCount.get(2);
+    }
+
+    /**
+     * get the number of info log.
+     * @return the number of info log
+     */
+    public static long getInfo() {
+        return sCount.get(3);
+    }
+
+    /**
+     * add the number of corresponding level log.
+     * @param event event of generating log
+     * @return
+     */
+    public void append(LoggingEvent event) {
```

Comment:
Who calls append? is the append directly called by log4j internally every time when a new log is logged?

Reply:
Yes, append is directly called by log4j every time a new log is generated